easybuild/framework/easyconfig/types.py

```diff
@@ -116,3 +111,64 @@ def convert_value_type(val, typ):
         raise EasyBuildError("No conversion function available (yet) for target type %s", typ)
 
     return res
+
+
+def to_toolchain(tcspec, convert_to='dict'):
+    """
+    Convert a toolchain string "intel, 2015a" to a dictionary {'name':'intel', 'version':'2015a'}
+
+    @param tcspec: a toolchain in the form of a string or a list
+    """
+    # check if tc is a string or a list of two values; else, it can not be converted
+    if isinstance(tcspec, basestring):
+        tcspec = tcspec.split(',')
+
+    if isinstance(tcspec, list):
+        if len(tcspec) == 2:
+            if convert_to == 'tuple':
+                res = (tcspec[0].strip(), tcspec[1].strip())
+            else:
+                res = {'name': tcspec[0].strip(), 'version': tcspec[1].strip()}
+        else:
+            errstr = "Can not convert list %s to name and version %s. Expected 2 elements"
+            raise EasyBuildError(errstr % (tcspec, convert_to))
+    else:
+        errstr = "Conversion of %s (type %s) to toolchain %s is not supported"
+        raise EasyBuildError(errstr % (tcspec, type(tcspec), convert_to))
+
+    return res
+
+
+def to_dependency(dep):
+    """
+    Convert a dependency dict from the new YAML format {'name': 'lib, 1.2.8', 'toolchain': 'gcc, 4.8.2'}
+    to a tuple in the dependency format
+    """
+    tup = ()
+    if isinstance(dep, dict):
+        if 'name' in dep:
+            tup += to_toolchain(dep['name'], convert_to='tuple')
+        else:
+            raise EasyBuildError("Can not parse dependency without name and version: %s", dep)
+
+        tup += (dep.get('versionsuffix', ''),)
+        if 'toolchain' in dep:
+            tcspec = to_toolchain(dep['toolchain'], convert_to='tuple')
+            tup += (tcspec,)
+        else:
+            tup += ('',)
```

Comment:
what's this about? empty versionsuffix?

the toolchain is the last entry in the tuple, so this doesn't make sense, you need to do this if there's no versionsuffix set